src/app/fddev/dev.c

```diff
@@ -61,12 +63,87 @@ install_parent_signals( void ) {
     FD_LOG_ERR(( "sigaction(SIGINT) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
 }
 
+static ushort
+compute_shred_version( char const * genesis_path ) {
+  /* Compute the shred version and the genesis hash */
+  fd_sha256_t _sha[ 1 ];  fd_sha256_t * sha = fd_sha256_join( fd_sha256_new( _sha ) );
+  fd_sha256_init( sha );
+  uchar buffer[ 1024 ];
+
+  FILE * genesis_file = fopen( genesis_path, "r" );
+  if( FD_UNLIKELY( !genesis_file ) ) {
+    if( FD_LIKELY( errno==ENOENT ) ) return (ushort)0;
+
+    FD_LOG_ERR(( "Opening genesis file (%s) failed (%i-%s)", genesis_path, errno, fd_io_strerror( errno ) ));
+  }
+
+  while( !feof( genesis_file ) ) {
+    ulong read = fread( buffer, 1UL, sizeof(buffer), genesis_file );
+    if( FD_UNLIKELY( ferror( genesis_file ) ) )
+      FD_LOG_ERR(( "fread failed `%s` (%i-%s)", genesis_path, errno, fd_io_strerror( errno ) ));
+
+    fd_sha256_append( sha, buffer, read );
+  }
+
+  if( FD_UNLIKELY( fclose( genesis_file ) ) )
+    FD_LOG_ERR(( "fclose failed `%s` (%i-%s)", genesis_path, errno, fd_io_strerror( errno ) ));
+
+  union {
+    uchar  c[ 32 ];
+    ushort s[ 16 ];
+  } hash;
+  fd_sha256_fini( sha, hash.c );
+  fd_sha256_delete( fd_sha256_leave( sha ) );
+
+  ushort xor = 0;
+  for( ulong i=0UL; i<16UL; i++ ) xor ^= hash.s[ i ];
+
+  xor = fd_ushort_bswap( xor );
+  return fd_ushort_if( xor<USHORT_MAX, (ushort)(xor + 1), USHORT_MAX );
+}
+
 void
 update_config_for_dev( config_t * const config ) {
   /* when starting from a new genesis block, this needs to be off else the
      validator will get stuck forever. */
   config->consensus.wait_for_vote_to_start_leader = 0;
 
+  /* We have to wait until we get a snapshot before we can join a second
+     validator to this one, so make this smaller than the default.  */
+  config->snapshots.full_snapshot_interval_slots = 200U;
+
+  /* Automatically compute the shred version from genesis if it
+     exists and we don't know it.  If it doesn't exist, we'll keep it
+     set to zero and get from gossip. */
+  if( FD_LIKELY( config->consensus.expected_shred_version==(ushort)0 ) ) {
```

Comment:
Why the cast here? I think the toml file supports ushort?

Reply:
The cast is on the 0 side to make it clear we're comparing two `ushort`s. If that's more confusing than helpful, I'll delete it.